node/bft/src/primary.rs

```diff
@@ -1081,6 +1105,46 @@ impl<N: Network> Primary<N> {
             }
         });
 
+        // Periodically try to increment to the next round.
+        // Note: This is necessary to ensure that the primary is not stuck on a previous round
+        // despite having received enough certificates to advance to the next round.
+        let self_ = self.clone();
+        self.spawn(async move {
+            loop {
+                // Sleep briefly.
+                tokio::time::sleep(Duration::from_millis(MAX_BATCH_DELAY_IN_MS)).await;
+                // If the primary is not synced, then do not increment to the next round.
+                if !self_.sync.is_synced() {
+                    trace!("Skipping round increment {}", "(node is syncing)".dimmed());
+                    continue;
+                }
+                // Attempt to increment to the next round.
+                let next_round = self_.current_round().saturating_add(1);
+                // Determine if the quorum threshold is reached for the current round.
+                let is_quorum_threshold_reached = {
+                    // Retrieve the certificates for the next round.
+                    let certificates = self_.storage.get_certificates_for_round(next_round);
+                    // If there are no certificates, then skip this check.
+                    if certificates.is_empty() {
+                        continue;
+                    }
+                    let authors = certificates.iter().map(BatchCertificate::author).collect();
```

Comment:
can we move this assignment under the committee lookback check? it would avoid an allocation every time that skip is triggered

Reply:
Resolved with [c53f5fd](https://github.com/AleoHQ/snarkOS/pull/3239/commits/c53f5fdc50cc47c68b04dad81e380e339e708e7b)